binary_tree_test.rb

```diff
@@ -0,0 +1,35 @@
+require 'minitest/autorun'
+require_relative 'binary_tree'
+
+class BinaryTreeTest < Minitest::Test
+  def test_empty_array
+    array=[]
+    empty = BinaryTree.new(nil,nil,"")
+    assert empty.left==nil&&empty.right==nil&&empty.value==nil.to_s, BinaryTree.create(array)
```

Comment:
Dobrze jest robić sprawdzenie jednej konkretnej rzeczy w jednym test casie. Gdybyśmy hipotetycznie dostali błąd w `test_empty_array` to nie wiedzielibyśmy od razu który z warunków nie jest spełniony, czyli dobrze byłoby rozbić ten test na drobniejsze (np. oddzielne testy na `left`, `right`, oraz `value`).

Dla lepszej czytelności do porównywania wartości używa się predefiniowanych asercji (matchers). Przykładowo testowanie czy wyrażenie `empty.left` równe jest `nil` wykonuje się przez `assert_nil(empty.left)`. Opis dostępnych metod: http://ruby-doc.org/stdlib-2.0.0/libdoc/minitest/rdoc/MiniTest/Assertions.html

 I najważniejsza sprawa - tak na prawdę te testy sprawdzają działanie inicjalizatora, który odpalany jest przez metodę `new`, a nie poprawność działania metody `create`. Zwróć uwagę, że jeśli zakomentujesz cały kod z metody `create` to nadal wszystkie testy przechodzą, a nie powinny.